### PR TITLE
fix(atsu): Fixes for NXApi-Freetext and CPDLC Logon-issues

### PR DESCRIPTION
**Pull request by svengcz**
## Summary of Changes

This PR fixes the freetext-issues based on the NXApi-telex protocol.
Additionally is NXApi marked as default freetext-protocol if Hoppie is disabled.

A last fix handles UNABLE responses much better on "REQUEST LOGON" message.

## Additional context

Discord username (if different from GitHub):

## Testing instructions

- Select FBW as freetext system and send yourself a telex-message -> check discord if message is sent -> wait for COMPANY MSG on upper ECAM -> read message in AOC/RECEIVED MESSAGES -> COMPANY MSG on upper ECAM removed
- Disable Hoppie on EFB -> Open Freetext in AOC menu -> expect FBW preselected -> Enable Hoppie -> Fill out INIT A -> Open Freetext in AOC menu -> Expect HOPPIES preselected
- Send REQUEST LOGON to ATC station -> NEXT STATION: XXXX on DCDU -> respond with UNABLE -> NEXT STATION: XXXX removed and Connection Notification is cleaned up after a reload

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Review by Saschl — COMMENTED** [review]
I’m no JS guru but afaik you either use ‘’’.then’’’ and resolve the promise as it was before, or ‘’’await’’‘ and then use the result directly.

File: src/atsu/src/com/webinterfaces/NXApiConnector.ts
Comment (Saschl) [inline src/atsu/src/com/webinterfaces/NXApiConnector.ts:205]: Something like:

```
// Fetch new messages
            try {
                const data = await Telex.fetchMessages();
                for (const msg of data) {
                    const message = new FreetextMessage();
                    message.Network = AtsuMessageNetwork.FBW;
                    message.Direction = AtsuMessageDirection.Uplink;
                    message.Station = msg.from.flight;
                    message.Message = msg.message.replace(/;/i, ' ');

                    retval.push(message);
                }
            } catch (e) {
                return [AtsuStatusCodes.ComFailed, retval];
            }
        }
```

Comment (svengcz) [reply src/atsu/src/com/webinterfaces/NXApiConnector.ts:205]: I don't see a real difference. Your version waits for the result of fetchMessages() and my version is waiting for the resulting array. Isn't it the same?

Comment (Saschl) [reply src/atsu/src/com/webinterfaces/NXApiConnector.ts:205]: I was just under the impression that either "await" or ".then" is being used. But I guess both works.

There could be an "issue" because as a result of "await" I want the result of the async operation. If I use ".then" in the async operation, the return value will be  ``undefined``.
 
It's not an issue now but just general practice I think. I won't block the PR due to this 👍

**Review by Saschl — APPROVED** [review]
Would be nice to unify the await/then usage but other than that LGTM.

**Comment by mico975** [comment]
Quality Assurance Tester Report

Discord          : mico#3145
Object of testing: #7174
Tier of Testing  : 1
Date             : 09/05/2022

Testing Process: 
Select FBW as freetext system and send yourself a telex-message -> check discord if message is sent -> wait for COMPANY MSG on upper ECAM -> read message in AOC/RECEIVED MESSAGES -> COMPANY MSG on upper ECAM removed - **OK**

Disable Hoppie on EFB -> Open Freetext in AOC menu -> expect FBW preselected -> Enable Hoppie -> Fill out INIT A -> Open Freetext in AOC menu -> Expect HOPPIES preselected - **OK**

Send REQUEST LOGON to ATC station -> NEXT STATION: XXXX on DCDU -> respond with UNABLE -> NEXT STATION: XXXX removed and Connection Notification is cleaned up after a reload  - **OK**

Negatives:       
After unable there's "notif failed" on notification page and I'm not sure if that's intended (see media).

Testing Results: 
Passed

Conclusions: 
LGTM

Media:
image

**Pull request merged**
